{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Carregar pacotes necessários\n",
    "from Bio import SeqIO   # Importar a função SeqIO do pacote Bio\n",
    "from Bio import AlignIO   # Importar a função AlignIO do pacote Bio\n",
    "from Bio import Entrez    # Importar a função Entrez do pacote Bio\n",
    "from collections import defaultdict   # Importar a função defaultdict do pacote collections\n",
    "import subprocess  # Importar a função subprocess\n",
    "import re   # Importar a função re\n",
    "import os    # Importar a função os\n",
    "\n",
    "# Configurar o e-mail para acessar o NCBI\n",
    "Entrez.email = \"nathanpaivab@gmail.com\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salvo: EU017577_EU018044_sequences/EU017577.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017578.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017579.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017580.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017581.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017582.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017583.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017584.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017585.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017586.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017587.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017588.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017589.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017590.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017591.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017592.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017593.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017594.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017595.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017596.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017597.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017598.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017599.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017600.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017601.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017602.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017603.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017604.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017605.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017606.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017607.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017608.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017609.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017610.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017611.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017612.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017613.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017614.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017615.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017616.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017617.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017618.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017619.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017620.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017621.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017622.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017623.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017624.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017625.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017626.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017627.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017628.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017629.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017630.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017631.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017632.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017633.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017634.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017635.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017636.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017637.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017638.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017639.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017640.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017641.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017642.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017643.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017644.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017645.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017646.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017647.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017648.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017649.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017650.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017651.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017652.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017653.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017654.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017655.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017656.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017657.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017658.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017659.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017660.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017661.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017662.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017663.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017664.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017665.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017666.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017667.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017668.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017669.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017670.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017671.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017672.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017673.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017674.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017675.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017676.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017677.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017678.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017679.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017680.1_Proceratophrys_boiei_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017681.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017682.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017683.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017684.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017685.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017686.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017687.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017688.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017689.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017690.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017691.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017692.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017693.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017694.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017695.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017696.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017697.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017698.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017699.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017700.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017701.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017702.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017703.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017704.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017705.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017706.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017707.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017708.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017709.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017710.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017711.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017712.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017713.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017714.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017715.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017716.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017717.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017718.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017719.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017720.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017721.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017722.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017723.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017724.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017725.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017726.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017727.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017728.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017729.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017730.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017731.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017732.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017733.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017734.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017735.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017736.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017737.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017738.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017739.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017740.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017741.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017742.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017743.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017744.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017745.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017746.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017747.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017748.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017749.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017750.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017751.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017752.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017753.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017754.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017755.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017756.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017757.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017758.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017759.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017760.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017761.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017762.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017763.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017764.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017765.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017766.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017767.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017768.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017769.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017770.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017771.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017772.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017773.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017774.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017775.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017776.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017777.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017778.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017779.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017780.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017781.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017782.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017783.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017784.1_Proceratophrys_boiei_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017785.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017786.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017787.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017788.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017789.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017790.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017791.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017792.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017793.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017794.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017795.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017796.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017797.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017798.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017799.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017800.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017801.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017802.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017803.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017804.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017805.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017806.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017807.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017808.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017809.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017810.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017811.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017812.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017813.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017814.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017815.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017816.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017817.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017818.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017819.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017820.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017821.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017822.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017823.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017824.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017825.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017826.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017827.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017828.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017829.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017830.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017831.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017832.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017833.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017834.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017835.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017836.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017837.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017838.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017839.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017840.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017841.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017842.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017843.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017844.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017845.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017846.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017847.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017848.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017849.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017850.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017851.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017852.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017853.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017854.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017855.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017856.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017857.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017858.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017859.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017860.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017861.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017862.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017863.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017864.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017865.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017866.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017867.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017868.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017869.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017870.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017871.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017872.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017873.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017874.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017875.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017876.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017877.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017878.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017879.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017880.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017881.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017882.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017883.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017884.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017885.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017886.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017887.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017888.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017889.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017890.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017891.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017892.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017893.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017894.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017895.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017896.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017897.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017898.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017899.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017900.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017901.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017902.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017903.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017904.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017905.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017906.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017907.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017908.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017909.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017910.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017911.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017912.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017913.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017914.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_cytb.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017915.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017916.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017917.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017918.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017919.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017920.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017921.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017922.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017923.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017924.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017925.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017926.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017927.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017928.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017929.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017930.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017931.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017932.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017933.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017934.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017935.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017936.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017937.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017938.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017939.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017940.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017941.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017942.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017943.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017944.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017945.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017946.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017947.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017948.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017949.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017950.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017951.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017952.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017953.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017954.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017955.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017956.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017957.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017958.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017959.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017960.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017961.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017962.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017963.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017964.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017965.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017966.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017967.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017968.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017969.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017970.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017971.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017972.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017973.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017974.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017975.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017976.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017977.1_Ischnocnema_aff._ramagii_ACC_JMB2007b_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017978.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017979.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017980.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017981.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017982.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017983.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017984.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017985.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017986.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017987.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017988.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017989.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017990.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017991.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017992.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017993.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017994.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017995.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017996.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017997.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017998.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU017999.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018000.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018001.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018002.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018003.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018004.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018005.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018006.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018007.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018008.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018009.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018010.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018011.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018012.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018013.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018014.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018015.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018016.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018017.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018018.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018019.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018020.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018021.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018022.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018023.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018024.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018025.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018026.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018027.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018028.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018029.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018030.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018031.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018032.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018033.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018034.1_Ischnocnema_aff._ramagii_ACC_JMB2007a_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018035.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018036.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018037.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018038.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018039.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018040.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018041.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018042.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018043.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n",
      "Salvo: EU017577_EU018044_sequences/EU018044.1_Ischnocnema_aff._ramagii_ACC_JMB2007c_ND2.fasta\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "def search_genes_ncbi_range(start_id, end_id):\n",
    "    \"\"\"\n",
    "    Pesquisa genes no NCBI a partir de um intervalo de códigos de identificação de genes.\n",
    "\n",
    "    Parameters:\n",
    "    start_id (str): Código de identificação do gene inicial no NCBI.\n",
    "    end_id (str): Código de identificação do gene final no NCBI.\n",
    "\n",
    "    Returns:\n",
    "    list: Lista de SeqRecord contendo as sequências dos genes.\n",
    "    \"\"\"\n",
    "    # Extrair prefixo e números dos IDs\n",
    "    prefix = re.match(r\"([A-Za-z]+)\", start_id).group(1)\n",
    "    start_num = int(re.match(r\"[A-Za-z]+(\\d+)\", start_id).group(1))\n",
    "    end_num = int(re.match(r\"[A-Za-z]+(\\d+)\", end_id).group(1))\n",
    "\n",
    "    gene_ids = [f\"{prefix}{i:06d}\" for i in range(start_num, end_num + 1)]\n",
    "\n",
    "    gene_records = []\n",
    "    for gene_id in gene_ids:\n",
    "        handle = Entrez.efetch(db=\"nucleotide\", id=gene_id, rettype=\"gb\", retmode=\"text\")\n",
    "        record = SeqIO.read(handle, \"genbank\")\n",
    "        handle.close()\n",
    "        gene_records.append(record)\n",
    "    return gene_records\n",
    "\n",
    "\n",
    "# Exemplo de uso\n",
    "start_id = \"EU017577\"\n",
    "end_id = \"EU018044\"\n",
    "\n",
    "\n",
    "gene_records = search_genes_ncbi_range(start_id, end_id)\n",
    "\n",
    "# Criar pasta para armazenar as sequências\n",
    "output_dir = f\"{start_id}_{end_id}_sequences\"\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "\n",
    "for record in gene_records:\n",
    "    # Extrair o nome da espécie e do gene do GenBank\n",
    "    species_name = record.annotations.get(\"organism\", \"Unknown_species\").replace(\" \", \"_\")\n",
    "    gene_name = \"Unknown_gene\"\n",
    "    for feature in record.features:\n",
    "        if feature.type == \"CDS\" and \"gene\" in feature.qualifiers:\n",
    "            gene_name = feature.qualifiers[\"gene\"][0]\n",
    "            break\n",
    "\n",
    "    # Salvar cada sequência em um arquivo separado com o formato especificado\n",
    "    output_path = os.path.join(output_dir, f\"{record.id}_{species_name}_{gene_name}.fasta\")\n",
    "    SeqIO.write(record, output_path, \"fasta\")\n",
    "    print(f\"Salvo: {output_path}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salvo: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_concatenated.fasta\n",
      "Salvo: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_concatenated.fasta\n"
     ]
    }
   ],
   "source": [
    "#concatenar sequências de genes\n",
    "\n",
    "# Diretório onde as sequências estão armazenadas\n",
    "input_dir = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences\"\n",
    "\n",
    "# Dicionário para armazenar sequências por gene\n",
    "gene_sequences = defaultdict(list)\n",
    "\n",
    "# Iterar sobre os arquivos no diretório\n",
    "for filename in os.listdir(input_dir):\n",
    "    if filename.endswith(\".fasta\"):\n",
    "        filepath = os.path.join(input_dir, filename)\n",
    "        record = SeqIO.read(filepath, \"fasta\")\n",
    "        gene_name = filename.split('_')[-1].replace(\".fasta\", \"\")\n",
    "        gene_sequences[gene_name].append(record)\n",
    "\n",
    "# Concatenar sequências de cada gene em um único arquivo\n",
    "for gene_name, records in gene_sequences.items():\n",
    "    output_path = os.path.join(input_dir, f\"{gene_name}_concatenated.fasta\")\n",
    "    SeqIO.write(records, output_path, \"fasta\")\n",
    "    print(f\"Salvo: {output_path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Salvo: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_concatenated.fasta\n"
     ]
    }
   ],
   "source": [
    "#concatenar sequências de genes\n",
    "\n",
    "# Diretório onde as sequências estão armazenadas\n",
    "input_dir = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences\"\n",
    "\n",
    "# Dicionário para armazenar sequências por gene\n",
    "gene_sequences = defaultdict(list)\n",
    "\n",
    "# Iterar sobre os arquivos no diretório\n",
    "for filename in os.listdir(input_dir):\n",
    "    if filename.endswith(\".fasta\"):\n",
    "        filepath = os.path.join(input_dir, filename)\n",
    "        record = SeqIO.read(filepath, \"fasta\")\n",
    "        gene_name = filename.split('_')[-1].replace(\".fasta\", \"\")\n",
    "        gene_sequences[gene_name].append(record)\n",
    "\n",
    "# Concatenar sequências de cada gene em um único arquivo\n",
    "for gene_name, records in gene_sequences.items():\n",
    "    output_path = os.path.join(input_dir, f\"{gene_name}_concatenated.fasta\")\n",
    "    SeqIO.write(records, output_path, \"fasta\")\n",
    "    print(f\"Salvo: {output_path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "ruby"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências filtradas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei.fasta\n"
     ]
    }
   ],
   "source": [
    "#filtrar sequências de NAD2 apenas de Proceratophrys boiei\n",
    "\n",
    "\n",
    "# Caminho do arquivo concatenado\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_concatenated.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo concatenado\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Filtrar sequências que pertencem à espécie Proceratophrys boiei\n",
    "filtered_sequences = [record for record in records if \"Proceratophrys boiei\" in record.description]\n",
    "\n",
    "# Salvar as sequências filtradas em um novo arquivo\n",
    "SeqIO.write(filtered_sequences, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências filtradas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "ruby"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências filtradas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei.fasta\n"
     ]
    }
   ],
   "source": [
    "#filtrar sequências de cytb apenas de Proceratophrys boiei\n",
    "\n",
    "# Caminho do arquivo concatenado\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_concatenated.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo concatenado\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Filtrar sequências que pertencem à espécie Proceratophrys boiei\n",
    "filtered_sequences = [record for record in records if \"Proceratophrys boiei\" in record.description]\n",
    "\n",
    "# Salvar as sequências filtradas em um novo arquivo\n",
    "SeqIO.write(filtered_sequences, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências filtradas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "ruby"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências filtradas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei.fasta\n"
     ]
    }
   ],
   "source": [
    "#filtrar sequências de c-myc apenas de Proceratophrys boiei\n",
    "\n",
    "# Caminho do arquivo concatenado\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_concatenated.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo concatenado\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Filtrar sequências que pertencem à espécie Proceratophrys boiei\n",
    "filtered_sequences = [record for record in records if \"Proceratophrys boiei\" in record.description]\n",
    "\n",
    "# Salvar as sequências filtradas em um novo arquivo\n",
    "SeqIO.write(filtered_sequences, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências filtradas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "ruby"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências renomeadas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_renamed.fasta\n"
     ]
    }
   ],
   "source": [
    "#renomear as sequências de ND2 de Proceratophrys boiei para que fique apenas o código e o quinto termo do cabeçalho\n",
    "\n",
    "\n",
    "# Caminho do arquivo de entrada e saída\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_renamed.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo de entrada\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Renomear os cabeçalhos das sequências\n",
    "for record in records:\n",
    "    description_parts = record.description.split()\n",
    "    sequence_id = description_parts[0]\n",
    "    fifth_term = description_parts[4] if len(description_parts) > 4 else \"\"\n",
    "    new_description = f\"{sequence_id} {fifth_term}\"\n",
    "    record.description = new_description\n",
    "    record.id = sequence_id  # Manter apenas o ID da sequência\n",
    "\n",
    "# Salvar as sequências com os novos cabeçalhos\n",
    "SeqIO.write(records, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências renomeadas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "ruby"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências renomeadas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_renamed.fasta\n"
     ]
    }
   ],
   "source": [
    "#renomear as sequências de cytb de Proceratophrys boiei para que fique apenas o código e o quinto termo do cabeçalho\n",
    "\n",
    "# Caminho do arquivo de entrada e saída\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_renamed.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo de entrada\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Renomear os cabeçalhos das sequências\n",
    "for record in records:\n",
    "    description_parts = record.description.split()\n",
    "    sequence_id = description_parts[0]\n",
    "    fifth_term = description_parts[4] if len(description_parts) > 4 else \"\"\n",
    "    new_description = f\"{sequence_id} {fifth_term}\"\n",
    "    record.description = new_description\n",
    "    record.id = sequence_id  # Manter apenas o ID da sequência\n",
    "\n",
    "# Salvar as sequências com os novos cabeçalhos\n",
    "SeqIO.write(records, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências renomeadas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "ruby"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências renomeadas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_renamed.fasta\n"
     ]
    }
   ],
   "source": [
    "#renomear as sequências de ND2 de Proceratophrys boiei para que fique apenas o código e o quinto termo do cabeçalho\n",
    "\n",
    "# Caminho do arquivo de entrada e saída\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_renamed.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo de entrada\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Renomear os cabeçalhos das sequências\n",
    "for record in records:\n",
    "    description_parts = record.description.split()\n",
    "    sequence_id = description_parts[0]\n",
    "    fifth_term = description_parts[4] if len(description_parts) > 4 else \"\"\n",
    "    new_description = f\"{sequence_id} {fifth_term}\"\n",
    "    record.description = new_description\n",
    "    record.id = sequence_id  # Manter apenas o ID da sequência\n",
    "\n",
    "# Salvar as sequências com os novos cabeçalhos\n",
    "SeqIO.write(records, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências renomeadas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "ruby"
    }
   },
   "outputs": [],
   "source": [
    "#caso queira as sequências de Proceratophrys boiei apenas com o número do voucher "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências renomeadas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_renamed_second_term.fasta\n"
     ]
    }
   ],
   "source": [
    "#caso queira as sequências de Proceratophrys boiei apenas com o número do voucher \n",
    "# Caminho do arquivo de entrada e saída\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_renamed.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_renamed_second_term.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo de entrada\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Renomear os cabeçalhos das sequências\n",
    "for record in records:\n",
    "    description_parts = record.description.split()\n",
    "    second_term = description_parts[1] if len(description_parts) > 1 else \"\"\n",
    "    new_description = f\"{second_term}\"\n",
    "    record.description = new_description\n",
    "    record.id = second_term  # Manter apenas o segundo termo como ID da sequência\n",
    "\n",
    "# Salvar as sequências com os novos cabeçalhos\n",
    "SeqIO.write(records, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências renomeadas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências renomeadas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_renamed_second_term.fasta\n"
     ]
    }
   ],
   "source": [
    "#caso queira as sequências de Proceratophrys boiei apenas com o número do voucher \n",
    "# Caminho do arquivo de entrada e saída\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_renamed.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_renamed_second_term.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo de entrada\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Renomear os cabeçalhos das sequências\n",
    "for record in records:\n",
    "    description_parts = record.description.split()\n",
    "    second_term = description_parts[1] if len(description_parts) > 1 else \"\"\n",
    "    new_description = f\"{second_term}\"\n",
    "    record.description = new_description\n",
    "    record.id = second_term  # Manter apenas o segundo termo como ID da sequência\n",
    "\n",
    "# Salvar as sequências com os novos cabeçalhos\n",
    "SeqIO.write(records, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências renomeadas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequências renomeadas salvas em: /home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_renamed_second_term.fasta\n"
     ]
    }
   ],
   "source": [
    "#caso queira as sequências de Proceratophrys boiei apenas com o número do voucher \n",
    "# Caminho do arquivo de entrada e saída\n",
    "input_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_renamed.fasta\"\n",
    "output_file = \"/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_renamed_second_term.fasta\"\n",
    "\n",
    "# Ler as sequências do arquivo de entrada\n",
    "records = list(SeqIO.parse(input_file, \"fasta\"))\n",
    "\n",
    "# Renomear os cabeçalhos das sequências\n",
    "for record in records:\n",
    "    description_parts = record.description.split()\n",
    "    second_term = description_parts[1] if len(description_parts) > 1 else \"\"\n",
    "    new_description = f\"{second_term}\"\n",
    "    record.description = new_description\n",
    "    record.id = second_term  # Manter apenas o segundo termo como ID da sequência\n",
    "\n",
    "# Salvar as sequências com os novos cabeçalhos\n",
    "SeqIO.write(records, output_file, \"fasta\")\n",
    "\n",
    "print(f\"Sequências renomeadas salvas em: {output_file}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "três arquivos com as sequências prontas para fazer o alinhamento foram organizados:\n",
    "\n",
    "/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_renamed_second_term.fasta\n",
    "/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_renamed_second_term.fasta\n",
    "/home/nathan/Documents/Doutorado_diversidade_genética/Filogeografia/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_renamed_second_term.fasta\n",
    "\n",
    "será feito o alinhamento dessas sequências"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "outputhat23=2\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "tbfast-pair (nuc) Version 7.490\n",
      "alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=2\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "  100 / 104\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP   103 /103 \n",
      "done.\n",
      "tbfast (nuc) Version 7.490\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "Loading 'hat3' ... done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "  100 / 104\n",
      "Segment   1/  1    1- 709\n",
      "STEP 002-102-1  identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.   \n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.490\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<2) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alinhamento concluído: /home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_aligned.fas\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "tbfast-pair (nuc) Version 7.490\n",
      "alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "   50 / 55\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP    54 /54 \n",
      "done.\n",
      "tbfast (nuc) Version 7.490\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "   50 / 55\n",
      "Segment   1/  1    1- 875\n",
      "STEP 002-053-1  identical.    identical.    identical.    identical.    identical.   \n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.490\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alinhamento concluído: /home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_aligned.fas\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "outputhat23=2\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "tbfast-pair (nuc) Version 7.490\n",
      "alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=2\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "  100 / 104\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP   103 /103 \n",
      "done.\n",
      "tbfast (nuc) Version 7.490\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 2\n",
      "sueff_global = 0.100000\n",
      "nadd = 2\n",
      "Loading 'hat3' ... done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "  100 / 104\n",
      "Segment   1/  1    1- 667\n",
      "STEP 001-098-0  identical.    identical.    identical.    identical.    identical.    identical.   "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alinhamento concluído: /home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_aligned.fas\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "STEP 002-102-1  identical.   \n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.490\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<2) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Lista de arquivos de entrada e saída\n",
    "input_files = [\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_renamed_second_term.fasta\",\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_renamed_second_term.fasta\",\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_renamed_second_term.fasta\"\n",
    "]\n",
    "\n",
    "output_files = [\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_aligned.fas\",\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_aligned.fas\",\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_aligned.fas\"\n",
    "]\n",
    "\n",
    "# Alinhar sequências usando MAFFT e salvar no formato .fas \n",
    "for input_file, output_file in zip(input_files, output_files):\n",
    "    command = [\"mafft\", \"--auto\", input_file]\n",
    "    with open(output_file, \"w\") as output_handle:\n",
    "        subprocess.run(command, stdout=output_handle)\n",
    "    print(f\"Alinhamento concluído: {output_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alinhamento do arquivo: /home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_aligned.fas\n",
      "Alignment with 104 rows and 708 columns\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BMD830\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BC767\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BC774\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att FC1402\n",
      "gttaaaccccctcctttactaattaacccgtatgccctatcaat...att SM1577\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att FC1065\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att TAP1480\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BC772\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BC728\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BMD840\n",
      "gttaaaccccctcctttactaattaacccgtatgccctatcaat...att SM1581\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BMD376\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att FC1404\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att FC1036\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BC775\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att TAP1483\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BMD375\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att AA936\n",
      "...\n",
      "gttaaaccccctcctttactaattaacccatatgccctatcaat...att BMD846\n",
      "\n",
      "\n",
      "Alinhamento do arquivo: /home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_aligned.fas\n",
      "Alignment with 55 rows and 874 columns\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa FC1015_2\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa BC726\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa SM1577\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa SM1580\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa AA925_2\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa SM1587\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa TAP1474_1\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa TAP1476\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa FC1038\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa AA936_1\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa AA936_2\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa SM1578\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa AA947_2\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa AA947_1\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa BC728\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa BC766\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa AA948\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa TAP1477_1\n",
      "...\n",
      "tggagaaagtggtgtcggaaaagctggcgtcctaccaggcttct...caa TAP1474_2\n",
      "\n",
      "\n",
      "Alinhamento do arquivo: /home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_aligned.fas\n",
      "Alignment with 104 rows and 666 columns\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac BMD826\n",
      "tttggctccctcctagggacatgccttattgtacaaattgttac...gac FC479\n",
      "tttggctccctcttaggaacatgccttattgtacaaattgttac...gac SM1580\n",
      "tttggctccctcctaggaacatgcctcattgtacaaattgttac...gac BC765\n",
      "tttggctccctcctagggacatgccttattgtacaaattgttac...gac TAP1521\n",
      "tttggctccctcctagggacatgccttattgtacaaattgttac...gac TAP1476\n",
      "tttggctccctcttaggaacatgccttattgtacaaattgttac...gac SM1586\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac BMD829\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac BMD830\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac FC1037\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac FC1404\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac BMD841\n",
      "tttggctccctcctaggaacatgcctcattgtacaaattgttac...gac BC730\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac BMD837\n",
      "tttggctccctcttaggaacatgccttattgtacaaattgttac...gac SM1588\n",
      "tttggctccctcctagggacatgccttattgtacaaattgttac...gac TAP1473\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac BMD828\n",
      "tttggctccctcctagggacatgcctcatcgtacaaattgttac...gac TAP1519\n",
      "...\n",
      "tttggctccctcctagggacatgcctcattgtacaaattgttac...gac FC1019\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Lista de arquivos de alinhamento\n",
    "alignment_files = [\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/ND2_Proceratophrys_boiei_aligned.fas\",\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU025551_EU025679_sequences/c-myc_Proceratophrys_boiei_aligned.fas\",\n",
    "    \"/home/nathan/Documents/Doutorado_diversidade_genética/Phylogeogra/EU017577_EU018044_sequences/cytb_Proceratophrys_boiei_aligned.fas\"\n",
    "]\n",
    "\n",
    "# Visualizar os alinhamentos\n",
    "for alignment_file in alignment_files:\n",
    "    alignment = AlignIO.read(alignment_file, \"fasta\")\n",
    "    print(f\"Alinhamento do arquivo: {alignment_file}\")\n",
    "    print(alignment)\n",
    "    print(\"\\n\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
