
<html>
<!--Arlequin results file-->

<head>
<style>
BODY {background-color: white; font-size: 8pt; font-family: Courier New}
</style>
</head>

<body bgcolor=white>

<PRE>

</data>

<header>
Arlequin results
</header>
<data>

<A NAME="22_11_24at23_50_00"></A>

////////////////////////////////////////////////////////////////////
RUN NUMBER 1 (22/11/24 at 23:50:00)
////////////////////////////////////////////////////////////////////

<A NAME="22_11_24at23_50_00_run_information"></A>

Project information:
--------------------
	NbSamples      =  2
	DataType       =  DNA
	GenotypicData  =  0

==============================
Settings used for Calculations
==============================

 General settings:
 -----------------
        Deletion Weight                = 1
        Transition Weight Weight       = 1
        Tranversion Weight Weight      = 1
        Epsilon Value                  = 1e-07
        Significant digits for output  = 5
        Infer haplotype definition from distance matrix
        Alllowed level of missing data = 0.05

 Active Tasks:
 -------------

    Standard indices:
    -----------------

    Molecular Diversity:
    --------------------
        Molecular Distance :Pairwise difference
        GammaA Value       = 0
        Theta estimators   : 
            Theta(Hom)
            Theta(S)
            Theta(k)
            Theta(Pi)

<A NAME="22_11_24at23_50_00_shared haplotypes"></A>

==================================================================
== Checking for haplotypes shared among populations:
==================================================================
<A NAME="22_11_24at23_50_00_gr_shared0"></A>
=====================================================================
== Population : PE_N_S
===============

   Haplotypes shared between "PE_N_S" and "PE_S_N":

		Hap_1 = Hap_1
		Hap_4 = Hap_4
		Hap_5 = Hap_5
		Hap_6 = Hap_6
		Hap_10 = Hap_10
		Hap_17 = Hap_17
		Hap_19 = Hap_19

---------------------------------------------------------------------

<A NAME="22_11_24at23_50_00_gr_shared1"></A>
=====================================================================
== Population : PE_S_N
===============

   Haplotypes shared between "PE_S_N" and "PE_N_S":

		Hap_1 = Hap_1
		Hap_4 = Hap_4
		Hap_5 = Hap_5
		Hap_6 = Hap_6
		Hap_10 = Hap_10
		Hap_17 = Hap_17
		Hap_19 = Hap_19

---------------------------------------------------------------------



Haplotype frequency distribution among the samples:
===================================================


Haplotype list
==============
Haplotype id                  Haplotype definition
------------                  --------------------
   Hap_1                      ATTGGTTGCCGCAAGGGG
   Hap_2                      GATCGTTGCCGCAAGGGG
   Hap_3                      GTTCGTTACCGTAAGGGG
   Hap_4                      GTTCGTTGCCGCAAGGGG
   Hap_5                      GTTGGTTGCCGCAAGGGG
   Hap_6                      ATTCGTTGCCGCAAGGGG
   Hap_7                      ATTCGTTGCAGCAAGGGA
   Hap_8                      ATTCGTTGCAGCAAGGGG
   Hap_9                      GTGGGTTGCCGCAAGGGG
   Hap_10                     GATGGTTGCCGCAAGGGG
   Hap_11                     ATTGGCAGACATAAGGGG
   Hap_12                     ATTCGTCGCTGCAAGGGG
   Hap_13                     GTTCGCAGCCGCAAGGGG
   Hap_14                     ATTCTTTGCCGCAAGGGG
   Hap_15                     GTTGGCTGCCGCAAGGGG
   Hap_16                     GTTGGCTGCCGCTCAGGG
   Hap_17                     GTTCGTTGCCGCAAGAGG
   Hap_18                     GATTGTTGCCGCAAGGGA
   Hap_19                     GTTGGTTGCCGCAAGGAG



Haplotype frequencies in populations
====================================

--------------------------------------
      Haplotype:     PE_N_S     PE_S_N
                       (40)       (44)
--------------------------------------
 Hap_1                19          1   
                                      
 Hap_2                 2          0   
                                      
 Hap_3                 1          0   
                                      
 Hap_4                 8         12   
                                      
 Hap_5                 3          1   
                                      
 Hap_6                 2          6   
                                      
 Hap_7                 0          2   
                                      
 Hap_8                 0          3   
                                      
 Hap_9                 0          1   
                                      
Hap_10                 1          1   
                                      
Hap_11                 0          4   
                                      
Hap_12                 0          1   
                                      
Hap_13                 0          1   
                                      
Hap_14                 0          3   
                                      
Hap_15                 0          3   
                                      
Hap_16                 0          1   
                                      
Hap_17                 3          2   
                                      
Hap_18                 0          1   
                                      
Hap_19                 1          1   
                                      


Relative frequencies:

      Haplotype:     PE_N_S     PE_S_N
                       (40)       (44)
--------------------------------------
 Hap_1             0.475     0.0227   
                                      
 Hap_2              0.05          0   
                                      
 Hap_3             0.025          0   
                                      
 Hap_4               0.2      0.273   
                                      
 Hap_5             0.075     0.0227   
                                      
 Hap_6              0.05      0.136   
                                      
 Hap_7                 0     0.0455   
                                      
 Hap_8                 0     0.0682   
                                      
 Hap_9                 0     0.0227   
                                      
Hap_10             0.025     0.0227   
                                      
Hap_11                 0     0.0909   
                                      
Hap_12                 0     0.0227   
                                      
Hap_13                 0     0.0227   
                                      
Hap_14                 0     0.0682   
                                      
Hap_15                 0     0.0682   
                                      
Hap_16                 0     0.0227   
                                      
Hap_17             0.075     0.0455   
                                      
Hap_18                 0     0.0227   
                                      
Hap_19             0.025     0.0227   
                                      

==============================================================================
== ANALYSES AT THE INTRA-POPULATION LEVEL
==============================================================================
<A NAME="22_11_24at23_50_00_group0"></A>

===============================================================================
== Sample : 	PE_N_S
===============================================================================


===============================
== Standard diversity indices : (PE_N_S)
===============================

Reference: <i>Nei, M., 1987.</i>
No. of gene copies       : 40
No. of sequences         : 9
No. of original 
sequences in sample      : 9
No. of loci              : 18
No. of usable loci       : 18 loci with less than  5.00 % missing data
No. of polymorphic sites : 7


Results are only shown for polymorphic loci
-------------------------------------------------
           Num.
           gene     Num.        Exp.
Locus#    copies   alleles       Het
-------------------------------------------------
     1        40         2   0.51154
     2        40         2   0.14231
     4        40         2   0.49231
     8        40         2   0.05000
    12        40         2   0.05000
    16        40         2   0.14231
    17        40         2   0.05000
-------------------------------------------------
  Mean    40.000     2.000   0.20549
  s.d.     0.000     0.000   0.20674
-------------------------------------------------


Haplotype-level computations
----------------------------

Sum of square freqs.     :  0.2837
Gene diversity           :  0.7346 +/-  0.0612

(Standard deviation is for the sampling process)

================================
== Molecular diversity indices : (PE_N_S)
================================

           <i>Tajima, F., 1983.</i>
           <i>Tajima, F. 1993.</i>
           <i>Nei, M., 1987.</i>
           <i>Zouros, E., 1979.</i>
           <i>Ewens, W.J. 1972.</i>
Sample size                                  : 40.0000
No. of haplotypes                            : 9
No. of original haplotypes in sample         : 9
Deletion weight                              : 1.0000
Transition weight                            : 1.0000
Transversion weight                          : 1.0000
Allowed level of missing data                : 5.0000 %
Number of observed transitions               : 5
Number of observed transversions             : 2
Number of substitutions                      : 7
Number of observed indels                    : 0
Number of polymorphic sites                  : 7
Number of observed sites with transitions    : 5
Number of observed sites with transversions  : 2
Number of observed sites with substitutions  : 7
Number of observed sites with indels         : 0
Number of observed nucleotide sites          : 18
Number of usable nucleotide sites            : 18

Nucleotide composition (Relative values)
   C : 18.75%
   T : 21.94%
   A : 15.14%
   G : 44.17%
   Total :100.00%


Distance method                              : Pairwise difference (no Gamma correction)

Mean number of pairwise differences          :     1.438462 +/-     0.894935

Nucleotide diversity (average over loci)     :     0.079915 +/-     0.055231

(Standard deviations are for both the sampling and the stochastic processes)

       Theta(Hom) :     2.137185
  S.D. Theta(Hom) :     0.711974

         Theta(k) :     3.311199
95 % confidence interval limits for theta(k) : [     1.531758,     6.821425 ]

         Theta(S) :     1.645687
    S.D. Theta(S) :     0.760020

        Theta(Pi) :     1.438462
   S.D. Theta(Pi) :     0.994161
<A NAME="22_11_24at23_50_00_group1"></A>

===============================================================================
== Sample : 	PE_S_N
===============================================================================


===============================
== Standard diversity indices : (PE_S_N)
===============================

Reference: <i>Nei, M., 1987.</i>
No. of gene copies       : 44
No. of sequences         : 17
No. of original 
sequences in sample      : 17
No. of loci              : 18
No. of usable loci       : 18 loci with less than  5.00 % missing data
No. of polymorphic sites : 17


Results are only shown for polymorphic loci
-------------------------------------------------
           Num.
           gene     Num.        Exp.
Locus#    copies   alleles       Het
-------------------------------------------------
     1        44         2   0.50740
     2        44         2   0.08879
     3        44         2   0.04545
     4        44         3   0.45772
     5        44         2   0.13002
     6        44         2   0.33298
     7        44         3   0.24630
     9        44         2   0.16913
    10        44         3   0.24630
    11        44         2   0.16913
    12        44         2   0.16913
    13        44         2   0.04545
    14        44         2   0.04545
    15        44         2   0.04545
    16        44         2   0.08879
    17        44         2   0.04545
    18        44         2   0.13002
-------------------------------------------------
  Mean    44.000     2.176   0.17429
  s.d.     0.000     0.393   0.14328
-------------------------------------------------


Haplotype-level computations
----------------------------

Sum of square freqs.     :  0.1240
Gene diversity           :  0.8964 +/-  0.0309

(Standard deviation is for the sampling process)

================================
== Molecular diversity indices : (PE_S_N)
================================

           <i>Tajima, F., 1983.</i>
           <i>Tajima, F. 1993.</i>
           <i>Nei, M., 1987.</i>
           <i>Zouros, E., 1979.</i>
           <i>Ewens, W.J. 1972.</i>
Sample size                                  : 44.0000
No. of haplotypes                            : 17
No. of original haplotypes in sample         : 17
Deletion weight                              : 1.0000
Transition weight                            : 1.0000
Transversion weight                          : 1.0000
Allowed level of missing data                : 5.0000 %
Number of observed transitions               : 11
Number of observed transversions             : 9
Number of substitutions                      : 20
Number of observed indels                    : 0
Number of polymorphic sites                  : 17
Number of observed sites with transitions    : 10
Number of observed sites with transversions  : 9
Number of observed sites with substitutions  : 17
Number of observed sites with indels         : 0
Number of observed nucleotide sites          : 18
Number of usable nucleotide sites            : 18

Nucleotide composition (Relative values)
   C : 20.08%
   T : 21.21%
   A : 16.79%
   G : 41.92%
   Total :100.00%


Distance method                              : Pairwise difference (no Gamma correction)

Mean number of pairwise differences          :     2.963002 +/-     1.580148

Nucleotide diversity (average over loci)     :     0.164611 +/-     0.097475

(Standard deviations are for both the sampling and the stochastic processes)

       Theta(Hom) :     7.382525
  S.D. Theta(Hom) :     2.699950

         Theta(k) :     9.680239
95 % confidence interval limits for theta(k) : [     5.213704,    17.659908 ]

         Theta(S) :     3.908047
    S.D. Theta(S) :     1.425916

        Theta(Pi) :     2.963002
   S.D. Theta(Pi) :     1.754541


================================================================================
== Summary of computations done within populations
================================================================================
<A NAME="22_11_24at23_50_00_comp_sum_Basic"></A>

----------------
Basic properties
--------------------------------------------------------------------
          Statistics      PE_N_S      PE_S_N        Mean        s.d.
--------------------------------------------------------------------
  No. of gene copies          40          44      42.000       2.828
         No. of loci          18          18      18.000       0.000
  No. of usable loci          18          18      18.000       0.000
  No. of polym. loci           7          17      12.000       7.071
--------------------------------------------------------------------
<A NAME="22_11_24at23_50_00_comp_sum_het"></A>

-----------------------
Expected heterozygosity
------------------------------------------------------------------
Locus#      PE_N_S      PE_S_N        Mean        s.d.   Tot. Het.
------------------------------------------------------------------
     1     0.51154     0.50740     0.50947     0.00293     0.50574
     2     0.14231     0.08879     0.11555     0.03784     0.11331
     3     0.00000     0.04545     0.02273     0.03214     0.02381
     4     0.49231     0.45772     0.47501     0.02446     0.51205
     5     0.00000     0.13002     0.06501     0.09194     0.06971
     6     0.00000     0.33298     0.16649     0.23545     0.19363
     7     0.00000     0.24630     0.12315     0.17416     0.13569
     8     0.05000     0.00000     0.02500     0.03536     0.02381
     9     0.00000     0.16913     0.08457     0.11960     0.09180
    10     0.00000     0.24630     0.12315     0.17416     0.13569
    11     0.00000     0.16913     0.08457     0.11960     0.09180
    12     0.05000     0.16913     0.10957     0.08424     0.11331
    13     0.00000     0.04545     0.02273     0.03214     0.02381
    14     0.00000     0.04545     0.02273     0.03214     0.02381
    15     0.00000     0.04545     0.02273     0.03214     0.02381
    16     0.14231     0.08879     0.11555     0.03784     0.11331
    17     0.05000     0.04545     0.04773     0.00321     0.04705
    18     0.00000     0.13002     0.06501     0.09194     0.06971
------------------------------------------------------------------
  Mean     0.07991     0.16461     0.12226     0.05989     0.12843
  s.d.     0.16035     0.14494     0.15265     0.01089     0.14663
------------------------------------------------------------------
<A NAME="22_11_24at23_50_00_comp_sum_numAll"></A>

-----------------
Number of alleles
------------------------------------------------------------------
Locus#      PE_N_S      PE_S_N        Mean        s.d. Tot. number
------------------------------------------------------------------
     1           2           2       2.000       0.000           2
     2           2           2       2.000       0.000           2
     3           1           2       1.500       0.707           2
     4           2           3       2.500       0.707           3
     5           1           2       1.500       0.707           2
     6           1           2       1.500       0.707           2
     7           1           3       2.000       1.414           3
     8           2           1       1.500       0.707           2
     9           1           2       1.500       0.707           2
    10           1           3       2.000       1.414           3
    11           1           2       1.500       0.707           2
    12           2           2       2.000       0.000           2
    13           1           2       1.500       0.707           2
    14           1           2       1.500       0.707           2
    15           1           2       1.500       0.707           2
    16           2           2       2.000       0.000           2
    17           2           2       2.000       0.000           2
    18           1           2       1.500       0.707           2
------------------------------------------------------------------
  Mean       1.389       2.111       1.750       0.511       2.167
  s.d.       0.502       0.471       0.487       0.021       0.383
------------------------------------------------------------------
<A NAME="22_11_24at23_50_00_comp_sum_moldiv"></A>

---------------------------
Molecular diversity indexes
-------------------------------------------------------------------------
               Statistics      PE_N_S      PE_S_N        Mean        s.d.
-------------------------------------------------------------------------
       No. of transitions           5          11       8.000       4.243
     No. of transversions           2           9       5.500       4.950
     No. of substitutions           7          20      13.500       9.192
            No. of indels           0           0       0.000       0.000
         No. of ts. sites           5          10       7.500       3.536
         No. of tv. sites           2           9       5.500       4.950
      No. of subst. sites           7          17      12.000       7.071  Total:          18
 No. private subst. sites           1          11       6.000       7.071
       No. of indel sites           0           0       0.000       0.000
                       Pi       1.438       2.963     2.20073     1.07801

                  Theta_k     3.31120     9.68024     6.49572     4.50359
            Theta_k_lower     1.53176     5.21370     3.37273     2.60353
            Theta_k_upper     6.82143    17.65991    12.24067     7.66396
                  Theta_H     2.13718     7.38252     4.75985     3.70902
             s.d. Theta_H     0.71197     2.69995     1.70596     1.40571
                  Theta_S     1.64569     3.90805     2.77687     1.59973
             s.d. Theta_S     0.76002     1.42592     1.09297     0.47086
                 Theta_pi     1.43846     2.96300     2.20073     1.07801
            s.d. Theta_pi     0.99416     1.75454     1.37435     0.53767
-------------------------------------------------------------------------

////////////////////////////////////////////////////////////////////
END OF RUN NUMBER 1 (22/11/24 at 23:50:00))
Total computing time for this run : 0h 0m 0s 67 ms
////////////////////////////////////////////////////////////////////


</body>
</PRE>
</html>

